### PR TITLE
feat: update project to Angular 16

### PR DESCRIPTION
**Pull request by michaelfaith**
The primary focus for this change is to update the project to Angular 16, verify functionality,
and update documentation to reflect compatibility.

In addition to this primary goal, I also did the following quality of life updates:
- removed unused libraries (e.g. protractor)
- migrated from `tslint` to `eslint` and resolved linting errors
- functional route guards

### CODE REVIEW

**Review by rowellsjr — COMMENTED** [review]
I see that Protractor and the corresponding test cases were removed, but I don't see that anything has been added to replace them. How are the test cases being covered otherwise?

**Comment by michaelfaith** [comment]
Protractor was set up, but wasn't being used.  So there isn't an alternative in there yet.

File: projects/quickstart-demo/e2e/src/app.e2e-spec.ts
Comment (rowellsjr) [inline projects/quickstart-demo/e2e/src/app.e2e-spec.ts:1]: Are these test cases needed?

**Comment by rowellsjr** [comment]
> Protractor was set up, but wasn't being used. So there isn't an alternative in there yet.

I saw a few cases that appeared to be using Protractor, but they were removed. I marked them for confirmation.

**Comment by michaelfaith** [comment]
> > Protractor was set up, but wasn't being used. So there isn't an alternative in there yet.
> 
> I saw a few cases that appeared to be using Protractor, but they were removed. I marked them for confirmation.

@rowellsjr  Those are just the boilerplate tests that came with a new angular app (before they stopped including protractor).  There wasn't anything meaningful to this repo in there.

**Comment by michaelfaith** [comment]
@rowellsjr I saw another comment from you here a moment ago, but it's not here anymore.  Did that answer your question?

**Comment by rowellsjr** [comment]
> @rowellsjr I saw another comment from you here a moment ago, but it's not here anymore. Did that answer your question?

Yes, that was due to a misunderstanding of GitHub's UI. I'm still learning. :)

**Comment by rowellsjr** [comment]
> > > Protractor was set up, but wasn't being used. So there isn't an alternative in there yet.
> > 
> > 
> > I saw a few cases that appeared to be using Protractor, but they were removed. I marked them for confirmation.
> 
> @rowellsjr Those are just the boilerplate tests that came with a new angular app (before they stopped including protractor). There wasn't anything meaningful to this repo in there.

Oh, okay. I just wanted to make sure.

**Review by rowellsjr — APPROVED** [review]

**Pull request merged**